File: monsoon3.0/src/main/java/wtf/monsoon/impl/module/movement/Sprint.java
package wtf.monsoon.impl.module.movement;

import io.github.nevalackin.homoBus.Listener;
import io.github.nevalackin.homoBus.annotations.EventLink;
import wtf.monsoon.Wrapper;
import wtf.monsoon.api.module.Category;
import wtf.monsoon.api.module.Module;
import wtf.monsoon.api.setting.Setting;
import wtf.monsoon.api.util.entity.MovementUtil;
import wtf.monsoon.impl.event.EventUpdate;
import wtf.monsoon.impl.module.player.Scaffold;

public class Sprint extends Module {

    public final Setting<Boolean> omni = new Setting<>("Omni", false)
            .describedBy("Whether to sprint in all directions.");

    public Sprint() {
        super("Sprint", "Always sprint", Category.MOVEMENT);
    }

    @Override
    public void onEnable() {
        super.onEnable();
    }

    @Override
    public void onDisable() {
        super.onDisable();
    }

    @EventLink
    public final Listener<EventUpdate> eventUpdateListener = e -> {
        if (player.isMoving() && !Wrapper.getModule(Scaffold.class).isEnabled()) {
            if (omni.getValue()) {
                mc.thePlayer.setSprinting(true);
            } else {
                if (player.isMoving() && (mc.thePlayer.moveForward >= Math.abs(mc.thePlayer.moveStrafing)))
                    mc.thePlayer.setSprinting(true);
            }
        }
    };
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/modules/movement/Sprint.kt
package wtf.monsoon.client.modules.movement

import me.bush.eventbuskotlin.EventListener
import me.bush.eventbuskotlin.listener
import wtf.monsoon.Wrapper
import wtf.monsoon.backend.Category
import wtf.monsoon.backend.module.Module
import wtf.monsoon.backend.setting.Setting
import wtf.monsoon.client.event.EventPacket
import wtf.monsoon.client.event.EventPreMotion
import wtf.monsoon.client.modules.player.Scaffold
import kotlin.math.abs

class Sprint : Module("Sprint", "Automatically sprint.", Category.MOVEMENT) {

    val omni = Setting("Omni", "Whether to sprint in all directions.", false)

    @EventListener
    val preMotion = fun(e: EventPreMotion) {
        if (player.moving && !(Wrapper.monsoon.getModule(Scaffold::class.java).isEnabled() && !Wrapper.monsoon.getModule(Scaffold::class.java).allowSprinting.getValue())) {
            if (omni.getValue()) {
                mc.thePlayer.isSprinting = true
            } else {
                if (player.moving && mc.thePlayer.moveForward >= abs(mc.thePlayer.moveStrafing)) mc.thePlayer.isSprinting = true
            }
        }
    }

}